extensions-utils/src/dependencies/call.rs

```diff
@@ -0,0 +1,43 @@
+use anyhow::Context;
+use fn_error_context::context;
+use std::{
+    ffi::OsStr,
+    path::Path,
+    process::{self, Command},
+};
+
+/// Calls a binary that was delivered with an extension tarball.
+///
+/// # Returns
+/// - On success, returns stdout as a string.
+/// - On error, returns an error message including stdout and stderr.
+#[context("Calling {} CLI failed, or, it returned an error.", binary_name)]
+pub fn call_extension_bundled_binary<S, I>(
+    dfx_cache_path: &Path,
+    binary_name: &str,
+    args: I,
+) -> anyhow::Result<String>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let extension_binary_path =
+        std::env::current_exe().map_err(|e| anyhow::anyhow!("Failed to get current exe: {}", e))?;
+    let extension_dir_path = extension_binary_path.parent().ok_or_else(|| {
+        anyhow::anyhow!(
+            "Failed to locate parent of dir of executable: {}",
+            extension_binary_path.display()
+        )
+    })?;
+    let binary_to_call = extension_dir_path.join(binary_name);
+    let mut command = Command::new(&binary_to_call);
+    // If extension's dependency calls dfx; it should call dfx in this dir.
+    command.env("PATH", dfx_cache_path.join("dfx"));
```

Comment:
I still don't understand removing the rest of the path.  This would make `dfx build/deploy` fail when trying to call `npm`, for one. `dfx extension run` doesn't do this, though it does pass `--dfx-cache-path`.

Could we leave the PATH alone here, and let the extension pass through `--dfx-cache-path` if it needs to?

Reply:
From rust docs https://doc.rust-lang.org/std/process/struct.Command.html#method.env
> pub fn [env](https://doc.rust-lang.org/std/process/struct.Command.html#method.env)<K, V>(&mut self, key: K, val: V) -> &mut [Command](https://doc.rust-lang.org/std/process/struct.Command.html)
...
Inserts or updates an explicit environment variable mapping.

but looks like I misinterpreted `updates` for `appends`, just tested it and indeed it overwrites the content of PATH which is not what I thought it will do

Reply:
From rust docs https://doc.rust-lang.org/std/process/struct.Command.html#method.env
> pub fn [env](https://doc.rust-lang.org/std/process/struct.Command.html#method.env)<K, V>(&mut self, key: K, val: V) -> &mut [Command](https://doc.rust-lang.org/std/process/struct.Command.html)
...
Inserts or updates an explicit environment variable mapping.

but looks like I misinterpreted `updates` for `appends`, just tested it and indeed it overwrites the content of PATH which is not what I thought it will do

Reply:
> Could we leave the PATH alone here, and let the extension pass through --dfx-cache-path if it needs to?

I think this is not entirely up to us, as it would require upstream changes in command-line interfaces of ic-admin ic-nns-init and sns-cli. The solution you're proposing is certainly more robust, but let's come back to this in different PR https://dfinity.atlassian.net/browse/SDK-1201